engines/terraform/resolve_test.go

```diff
@@ -0,0 +1,200 @@
+package terraform
+
+import (
+	"testing"
+
+	"github.com/aws/jsii-runtime-go"
+	"github.com/hashicorp/terraform-cdk-go/cdktf"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSpecReferenceFromToken(t *testing.T) {
+	tests := []struct {
+		name        string
+		token       string
+		expected    *SpecReference
+		expectError bool
+		errorMsg    string
+	}{
+		{
+			name:  "valid infra reference",
+			token: "${infra.resource.property}",
+			expected: &SpecReference{
+				Source: "infra",
+				Path:   []string{"resource", "property"},
+			},
+			expectError: false,
+		},
+		{
+			name:  "valid self reference",
+			token: "${self.my_var}",
+			expected: &SpecReference{
+				Source: "self",
+				Path:   []string{"my_var"},
+			},
+			expectError: false,
+		},
+		{
+			name:  "valid self reference with nested path",
+			token: "${self.my_var.nested.value}",
+			expected: &SpecReference{
+				Source: "self",
+				Path:   []string{"my_var", "nested", "value"},
+			},
+			expectError: false,
+		},
+		{
+			name:  "valid var reference",
+			token: "${var.platform_var}",
+			expected: &SpecReference{
+				Source: "var",
+				Path:   []string{"platform_var"},
+			},
+			expectError: false,
+		},
+		{
+			name:        "invalid token format - no closing brace",
+			token:       "${infra.resource",
+			expected:    nil,
+			expectError: true,
+			errorMsg:    "invalid reference format",
+		},
+		{
+			name:        "invalid token format - not enough parts",
+			token:       "${infra}",
+			expected:    nil,
+			expectError: true,
+			errorMsg:    "invalid reference format",
+		},
+		{
+			name:        "invalid token format - only source",
+			token:       "${self}",
+			expected:    nil,
+			expectError: true,
+			errorMsg:    "invalid reference format",
+		},
```

Comment:
Add the equivalent test for `var` references. Also, can you include tests where the dot is include, but not the path value, e.g. `${self.}`